Tests/SonarScanner.MSBuild.Shim.Test/PropertiesFileGeneratorTests.cs

```diff
@@ -1457,6 +1457,54 @@ public void GenerateFile_AdditionalFiles_OnlyTestFiles_EndToEnd()
             .Contain($"sonar.tests=\\{Environment.NewLine}{string.Join($",\\{Environment.NewLine}", testFiles.Select(x => x.Replace("\\", "\\\\")))}");
     }
 
+    [DataTestMethod]
+    [DataRow("https://sonarcloud.io")]
+    [DataRow("https://sonarqube.us")]
+    [DataRow("https://sonarqqqq.whale")]    // Any value, as long as it was auto-computed by the default URL mechanism and stored in SonarQubeAnalysisConfig.xml
+    public void TryWriteProperties_HostUrl_NotSet_UseSonarQubeHostUrl(string sonarQubeHostUrl)
+    {
+        var outPath = Path.Combine(TestContext.TestRunDirectory, ".sonarqube", "out");
+        var config = new AnalysisConfig { SonarProjectKey = "key", SonarOutputDir = outPath, SonarQubeHostUrl = sonarQubeHostUrl };
+        var writer = new PropertiesWriter(config, logger);
+        TryWriteProperties_HostUrl_Execute(config, writer);
+
+        writer.Flush().Should().Contain($"sonar.host.url={sonarQubeHostUrl}");
+        logger.AssertDebugLogged("Setting analysis property: sonar.host.url=" + sonarQubeHostUrl);
+    }
+
+    [TestMethod]
+    public void TryWriteProperties_HostUrl_ExplicitValue_Propagated()
+    {
+        var outPath = Path.Combine(TestContext.TestRunDirectory, ".sonarqube", "out");
+        var config = new AnalysisConfig { SonarProjectKey = "key", SonarOutputDir = outPath, SonarQubeHostUrl = "Property should take precedence and this should not be used" };
+        config.LocalSettings = [new Property("sonar.host.url", "http://localhost:9000")];
+        var writer = new PropertiesWriter(config, logger);
+        TryWriteProperties_HostUrl_Execute(config, writer);
+
+        writer.Flush().Should().Contain("sonar.host.url=http://localhost:9000");
+    }
+
+    public void TryWriteProperties_HostUrl_Execute(AnalysisConfig config, PropertiesWriter writer)
+    {
+        Directory.CreateDirectory(config.SonarOutputDir);
+        var sut = new PropertiesFileGenerator(config, logger);
+        var projectPath = TestUtils.CreateEmptyFile(config.SonarOutputDir, "Project.csproj");
```

Comment:
The `TestUtils` methods create a lot of files in the test directory. I'm not sure if this really a good idea because it leaves files behind. We should either:
* Keep as is
* Use `TempFile` and similar types that delete the file on Dispose()
* Use the IFileWrapper/IDirectoyWrapper

Reply:
I agree we should improve the test framework, but not in this PR